ConcordiumWallet/Views/Stake/Baking/AmountInput/BakerAmountInputPresenter.swift

```diff
@@ -149,7 +149,13 @@ class BakerAmountInputPresenter: StakeAmountInputPresenterProtocol {
             guard let self = self else {
                 return .failure(StakeError.internalError)
             }
-            
+        
+            if let stakedAmount = account.baker?.stakedAmount  {
+                if amount == GTU(intValue: stakedAmount) {
+                    return .success(amount)
+                }
+            }
+        
```

Comment:
```suggestion
            if let baker = account.baker, baker.stakedAmount == amount.intValue {
                return .success(amount)
            }
```
is slightly cleaner IMHO.

Reply:
Also consider adding a comment explaining why short circuit is done.